src/transformerbeemcp/rest_api.py

```diff
@@ -0,0 +1,156 @@
+"""REST API for EDIFACT summarization."""
+
+import logging
+import os
+from collections import defaultdict
+from time import time
+
+import jwt
+from fastapi import Depends, FastAPI, HTTPException
+from fastapi.middleware.cors import CORSMiddleware
+from fastapi.security import HTTPAuthorizationCredentials, HTTPBearer
+from jwt import PyJWKClient
+from pydantic import BaseModel
+
+from .summarizer import summarize_edifact
+
+_logger = logging.getLogger(__name__)
+
+# Auth0 configuration
+AUTH0_DOMAIN = os.getenv("AUTH0_DOMAIN", "hochfrequenz.eu.auth0.com")
+AUTH0_AUDIENCE = os.getenv("AUTH0_AUDIENCE", "https://transformer.bee")
+JWKS_URL = f"https://{AUTH0_DOMAIN}/.well-known/jwks.json"
+
+# CORS configuration
+ALLOWED_ORIGINS = os.getenv(
+    "ALLOWED_ORIGINS",
+    "http://localhost:5173,https://nice-mushroom-04ebea203.3.azurestaticapps.net,https://thankful-water-00644131e.3.azurestaticapps.net",
+).split(",")
+
+# Rate limiting: requests per minute per user
+RATE_LIMIT = int(os.getenv("RATE_LIMIT", "10"))
+RATE_WINDOW = int(os.getenv("RATE_WINDOW", "60"))  # seconds
+rate_limit_store: dict[str, list[float]] = defaultdict(list)
+
+app = FastAPI(
+    title="TransformerBee Summarizer",
+    description="REST API for EDIFACT summarization using local LLM",
+    version="0.1.0",
```

Comment:
the version should be read from a version file which is overwritten in the build workflow.
check these:
1. a github action build step calls a script https://github.com/Hochfrequenz/fristenkalender-functions/blob/a36616b960b7368c77aceaaeea60af3f3b18683f/.github/workflows/publish-ghcr.yml#L19-L22
2. the script overwrites a version.json file https://github.com/Hochfrequenz/fristenkalender-functions/blob/main/scripts/update_version.sh
3. the version file is read: https://github.com/Hochfrequenz/fristenkalender-functions/blob/a36616b960b7368c77aceaaeea60af3f3b18683f/src/app/main.py#L12-L26
4. and its value is used to have the verion info in the app: https://github.com/Hochfrequenz/fristenkalender-functions/blob/a36616b960b7368c77aceaaeea60af3f3b18683f/src/app/main.py#L51-L54

Reply:
ok, letstry your approach